doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,167 +866,137 @@ Module system
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Module N : SIG. … End N.
 
-.. cmd:: Include @module
+  hints and the like valid for ``N`` are not those defined in ``M``
+  (or the module body) but the ones defined in ``SIG``.
 
-   Same as ``Include`` inside a module.
+.. _terminating_module:
 
-.. cmd:: Include {+<+ @module}
+Terminating an interactive module or module type definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
+.. todo this size is overkill for such a short section
 
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
+.. todo: description below applies equally to modules and module types?
 
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
+Interactive modules are terminated with the :cmd:`End` command, which
+is also used to terminate :ref:`Sections<section-mechanism>`.
+:n:`End @ident` closes the interactive module or module type :token:`ident`.
+If the module type was given, the command verifies that the content of the module
+matches the module type.  If the module is basic (not a
+functor), its components (constants, inductive types, submodules etc.)
+are now available through the dot notation.
 
-.. cmd:: End @ident
+.. exn:: No such label @ident.
+    :undocumented:
 
-   This command closes the interactive module type :token:`ident`.
+.. exn:: Signature components for label @ident do not match.
+    :undocumented:
 
-   .. exn:: This is not the last opened module type.
-      :undocumented:
+.. exn:: This is not the last opened module.
+    :undocumented:
 
-.. cmd:: Module Type @ident := @module_type
+.. exn:: This is not the last opened module type.
+   :undocumented:
 
-   Defines a module type :token:`ident` equal to :token:`module_type`.
+.. cmd:: Include @module_type_inl {* <+ @module_expr_inl }
 
-    .. cmdv:: Module Type @ident {* @module_binding} := @module_type
+   Includes the content of module(s) in the current
+   interactive module. Here :n:`@module_type_inl` can be a module expression or a module
+   type expression. If it is a high-order module or module type
+   expression then the system tries to instantiate :n:`@module_type_inl` with the current
+   interactive module.
 
-       Defines a functor type :token:`ident` specifying functors taking arguments :token:`module_bindings` and
-       returning :token:`module_type`.
+   Including multiple modules is a single :cmd:`Include` is equivalent to including each module
+   in a separate :cmd:`Include` command.
 
-    .. cmdv:: Module Type @ident {* @module_binding} := {+<+ @module_type }
+.. cmd:: Include Type @module_type_inl {* <+ @module_type_inl }
 
-       is equivalent to an interactive module type were each :token:`module_type` is included.
+   .. deprecated:: Use :cmd:`Include` instead.
```

Comment:
Actually `deprecated` wants to know since which version, so:

```suggestion
   .. deprecated:: 8.3
   
      Use :cmd:`Include` instead.
```

But since this has been deprecated and undocumented for so long, the best thing to do is probably to just remove it. See #11876.

Reply:
I would remove it when the command is removed.  Otherwise it will show up (to me, at least) as a undocumented command.  And it will be obvious to me when the command has been removed.